sycl/include/sycl/detail/handler_proxy.hpp

```diff
@@ -20,8 +20,15 @@ namespace detail {
 
 class AccessorBaseHost;
 
+#ifndef __SYCL_DEVICE_ONLY__
```

Comment:
Minor readability improvement: when there is an else branch, it's better to avoid negation. I.e. prefer
```c++
#ifdef
X
#else
Y
#endif
```
to
```c++
#ifndef
Y
#else
X
#endif
```